src/main/java/decodex/parser/Parser.java

```diff
@@ -1,5 +1,75 @@
 package decodex.parser;
 
+import decodex.commands.Command;
+import decodex.commands.ExitCommand;
+import decodex.data.DataManager;
+import decodex.modules.ModuleManager;
+import decodex.parser.exception.ParserException;
+import decodex.ui.Ui;
+
+import java.util.Arrays;
+
+/**
+ * Parses and validates the user input.
+ */
 public class Parser {
 
+    /**
+     * Returns the type of command that user has entered.
+     *
+     * @param userInput The input specified by the user.
+     * @return The type of command.
+     */
+    public String getCommandType(String userInput) throws ParserException {
+        String[] tokens = userInput.split(" ");
+        String commandType = tokens[0];
+        if (commandType.isEmpty()) {
+            throw new ParserException(ParserException.MISSING_COMMAND_TYPE_MESSAGE);
+        }
+        return tokens[0];

```

Comment:
Perhaps instead of checking if `commandType` is empty, I think you should check whether `tokens` is empty? If `tokens` is empty, then we will throw the exception.

Reply:
Perhaps doing a `strip()` on the `userInput` first would be better?